src/client/src/Client.c

```diff
@@ -50,6 +50,7 @@ STATUS checkIntermittentProducerCallback(UINT32 timerId, UINT64 currentTime, UIN
     pKinesisVideoClient->clientCallbacks.unlockMutexFn(pKinesisVideoClient->clientCallbacks.customData, pKinesisVideoClient->base.lock);
 
     if (STATUS_SUCCEEDED(retStatus)) {
+        pKinesisVideoClient->clientCallbacks.lockMutexFn(pKinesisVideoClient->clientCallbacks.customData, pKinesisVideoClient->base.putFrameLock);

```

Comment:
Note this is a client level lock. So it is what some may call a "giant lock", but for correctness (avoiding deadlock) in the present state of the code, this is necessary because in the intermittent producer callback we actually need to acquire this lock but if it's a stream level lock we wouldn't have access to it until after we iterate over the streams (which requires we in advance have the `streamListLock`) and if we have that then we can't *first* acquire this putFrame lock it needs to be acquired *before* acquiring the `streamListLock` and it's impossible to do this.